client/cache/MyCache.cs

```diff
@@ -65,21 +65,22 @@ public bool ContainsKey(TK key)
     internal class MemoryCacheWithCounter<TK, TV> : MemoryCache<TK, TV>
     {
         private int itemCount = 0;
-        
+
         public new void Put(TK key, TV value)
```

Comment:
Two writes were happening here. Instead we use one write and if the key was added we increment the limits counter.